src/toGamut.js

```diff
@@ -46,26 +81,59 @@ export default function toGamut (color, { method = defaults.gamut_mapping, space
 		}
 
 		if (method !== "clip" && !inGamut(color, space)) {
+
+			if (Object.prototype.hasOwnProperty.call(GMAPPRESET, method)) {
+				({method, jnd, deltaEMethod, blackWhiteClamp} = GMAPPRESET[method]);
+			}
+
+			// Get the correct delta E method
+			let de = deltaE2000;
+			if (deltaEMethod !== "") {
+				for (let m in deltaEMethods) {
+					if ("deltae" + deltaEMethod.toLowerCase() === m.toLowerCase()) {
+						de = deltaEMethods[m];
+						break;
+					}
+				}
+			}
+
 			let clipped = toGamut(clone(spaceColor), { method: "clip", space });
-			if (deltaE2000(color, clipped) > 2) {
+			if (de(color, clipped) > jnd) {
+
+				// Clamp to SDR white and black if required
+				if (Object.keys(blackWhiteClamp).length > 0) {
+					let channelMeta = ColorSpace.resolveCoord(blackWhiteClamp.channel);
+					let channel = get(to(color, channelMeta.space), channelMeta.id);
+					if (util.isNone(channel)) {
+						channel = 0;
+					}
+					if (channel >= blackWhiteClamp.max) {
+						return to({ space: "xyz-d65", coords: WHITES["D65"] }, color.space);
+					}
+					else if (channel <= blackWhiteClamp.min) {
+						return to({ space: "xyz-d65", coords: [0, 0, 0] }, color.space);
+					}
+				}
+
 				// Reduce a coordinate of a certain color space until the color is in gamut
 				let coordMeta = ColorSpace.resolveCoord(method);
 				let mapSpace = coordMeta.space;
 				let coordId = coordMeta.id;
 
-				let mappedColor = to(spaceColor, mapSpace);
+				let mappedColor = to(color, mapSpace);
 				let bounds = coordMeta.range || coordMeta.refRange;
 				let min = bounds[0];
-				let ε = .01; // for deltaE
+				let ε = calcEpsilon(jnd);
+				let minConverge = Math.max(ε, 1e-6); // limit delta of low and high to prevent infinite looping
```

Comment:
Why do we need a separate variable here? Can't we apply the clamping to ε directly?

Reply:
Do you prefer I limit the JND? In the end, I need to place some limit on size so we never enter an infinite loop. To be honest, I don't know why the same epsilon is used in these two cases as they are measuring different things.

Reply:
To be clear, I can use the same value for both of today is what is desired. Is my arbitrary limit satisfactory?

Reply:
I went ahead and combined them.